Gems/Terrain/Code/Source/Components/TerrainSystemComponent.cpp

```diff
@@ -63,6 +67,10 @@ namespace Terrain
         // every time an entity is added or removed to a level.  If this ever changes, the Terrain System ownership could move into
         // the level component.
         m_terrainSystem = new TerrainSystem();
+
+        auto* passSystem = AZ::RPI::PassSystemInterface::Get();
+        AZ_Assert(passSystem, "Cannot get the pass system.");
+        passSystem->AddPassCreator(AZ::Name("ClipmapGenerationComputePass"), &ClipmapGenerationPass::Create);
```

Comment:
Should this be in the Feature Processor's Activate()?  And should there be a corresponding call to remove the pass in a Deactivate?

Reply:
All other AddPassCreator happen in some system components rather than the feature processor. Let me get more info to see if can do it in that way.

Reply:
Unfortunately this one is doomed to be defined inside GetProvidedServices() in hopes to guarantee that the render pipeline is initialized after AddPassCreator() is called.